enterprise_learner_portal/api/v1/views.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+
+from django.contrib.auth import get_user_model
+from django.shortcuts import get_object_or_404
+from django.utils.translation import ugettext as _
+
+from rest_framework import permissions
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from edx_rest_framework_extensions.auth.bearer.authentication import BearerAuthentication
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+
+from enterprise.models import EnterpriseCourseEnrollment, EnterpriseCustomerUser
+from enterprise.utils import NotConnectedToOpenEdX
+from enterprise_learner_portal.api.v1.serializers import EnterpriseCourseEnrollmentSerializer
+
+try:
+    from openedx.core.djangoapps.content.course_overviews.api import get_course_overviews
+except ImportError:
+    get_course_overviews = None
+
+User = get_user_model()
+
+
+class EnterpriseCourseEnrollmentView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (JwtAuthentication, BearerAuthentication, SessionAuthentication,)
+
+    def get(self, request, user_email):
+        if get_course_overviews is None:
+            raise NotConnectedToOpenEdX(
+                _('To use this endpoint, this package must be '
+                  'installed in an Open edX environment.')
+            )
+
+        user = get_object_or_404(User, email=user_email)
```

Comment:
I think we want this endpoint to return the enrollments for the authenticated user. The client should not be able to request a specific user with a query param.

Reply:
Is your concern that someone viewing a page on the frontend could get information about other learners?

Reply:
Yes, that would be a concern. We'd need to add an is_staff check in here to allow someone to request another user's enrollments. In general, though, I think this endpoint is designed to service the learner portal and in that context you would never need to request another user's enrollments.

Reply:
Okay. I can definitely make sure `is_staff` is a requirement.

As for the resource the endpoint uses to return data, the endpoint itself is more flexible if we don't limit it to the authenticated user. You could imagine a staff user having the option of checking data for number of users other than one's own. Especially where this is a new endpoint, I don't think we need to add this restriction just yet